Services/TMS/Mailing/classes/class.ilTMSICalBuilder.php

```diff
@@ -0,0 +1,146 @@
+<?php
+use ILIAS\TMS\Mailing;
+
+require_once('./Services/TMS/Mailing/classes/class.ilTMSMailContextCourse.php');
+
+/**
+ * Class ilTMSICalBuilder
+ *
+ * @author Daniel Weise <daniel.weise@concepts-and-training.de>
+ * @copyright Extended GPL, see LICENSE
+ */
+class ilTMSICalBuilder implements Mailing\ICalBuilder
+{
+	const TITLE = "Titel";
+	const DESCRIPTION = "Beschreibung";
+	const DATE = "Datum";
+	const VENUE = "Veranstalter";
+	const TIME = "Zeit";
+	const FILE_EXTENSION = ".ics";
+
+	/**
+	 * @inheritdoc
+	 */
+	public function getICalString(array $info)
+	{
+		return $this->buildICal($info);
+	}
+
+	/**
+	 * Build an iCal string.
+	 *
+	 * @param 	array 	$info
+	 * @return 	string
+	 */
+	protected function buildICal(array $info)
+	{
+		$crs_name = "";
+		$description = "";
+		$duration = "";
+		$times = "";
+		$venue = "";
+
+		foreach($info as $i) {
+			switch ($i->getLabel()) {
+				case self::TITLE:
```

Comment:
Das wird nur so lange klappen, wie das TMS auf deutsch läuft. Danach geht es in die Brüche.
Am besten du machst mal nen Ticket dafür auf.